src/scripts/towns/TownList.ts

```diff
@@ -2864,6 +2867,166 @@ const LucyStevens1 = new NPC('Lucy Stevens', [
+    '...Eh, it was a Happiny? ...That\'s one fast Happiny, then, isn\'t it?',
+], {
+    image: 'assets/images/npcs/Waitress.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 15), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 17, GameConstants.AchievementOption.less)]),
+});
+
+const HappinyWitness5 = new NPC('Pokémon Ranger', [
+    'Ah, you\'re the one chasing after that Happiny? You look a bit puffed, are you okay?',
+    'I tried to capture it, but it barrelled right through my Capture Disc\'s trail! It went for Hearthome City. It wasn\'t showing any signs of slowing down yet.',
+], {
+    image: 'assets/images/npcs/Pokemon Ranger (female).png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 16), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 18, GameConstants.AchievementOption.less)]),
+});
+
+const HappinyWitness6 = new NPC('Murph', [
+    'Me and my buddy Slowpoke are trying to catch a Happiny that just went running into Mt. Coronet.',
+    'It had a blue egg on it\'s belly - I\'ve definitely seen an egg like that before, back in Fiore.',
+    'My Slowpoke\'s gone ahead, I hope I can catch up to them soon.',
+], {
+    image: 'assets/images/npcs/Murph.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 17), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 19, GameConstants.AchievementOption.less)]),
+});
+
+const HappinyWitness7 = new NPC('Cyclist', [
+    'Oh hey! Still chasing that Happiny, huh?',
+    'When I saw it still running, I tried to catch up to it, but it got into the Oreburgh Tunnel and I lost track of it.',
+    '...Say, are you alright? You look a little... bug-eyed.',
+    '<i>...This annoying little twerp has been taking you in one big loop, hasn\'t it...</i>',
+], {
+    image: 'assets/images/trainers/Cyclist (female).png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 18), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 20, GameConstants.AchievementOption.less)]),
+});
+
+const HappinyWitness8 = new NPC('Bug Catcher', [
+    'Some guy in a weird white outfit was chasing a Happiny!',
+    'They were heading south, towards Professor Rowan\'s lab!',
+    'You\'re chasing after them too, right? Good luck catching up to them!',
+], {
+    image: 'assets/images/trainers/Bug Catcher.png',
+    requirement: new MultiRequirement([new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 19), new QuestLineStepCompletedRequirement('Recover the Precious Egg!', 21, GameConstants.AchievementOption.less)]),
+});
+
+const HappinyWitness9 = new NPC('Go-Rock Pincher', [
+    'Damn, that Happiny got away. Those little friends got me good.',
```

Comment:
What are "Those little friends"?

Reply:
"those little friends" are the Pichu and Glameow that join Happiny. THey're from the Shadows of Almia questline. 
I'll reword the quest to have them explained a bit more clearly.